### PR TITLE
fix!: predicate cost estimation

### PR DESCRIPTION
**Pull request by hal3e**
close: https://github.com/FuelLabs/fuels-rs/issues/1123
codeveloped with: @segfault-magnet, @MujkicA and @Salka1988 

### Checklist
- [x] I have linked to any relevant issues.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary labels.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: packages/fuels/tests/providers.rs [hunk 1]
```diff
@@ -517,8 +517,9 @@ async fn test_gas_errors() -> Result<()> {
         .await
         .expect_err("should error");
 
-    let expected = "Provider error: Response errors; not enough coins to fit the target";
-    assert!(response.to_string().starts_with(expected));
+    // let expected = "Provider error: Response errors; not enough coins to fit the target";
+    dbg!(&response.to_string());
```
Comment (digorithm) [inline packages/fuels/tests/providers.rs:521]: Is this a leftover `dbg!` statement? Also, the `assert!` is commented out -- is it by design?

File: packages/fuels-accounts/src/accounts_utils.rs [hunk 2]
```diff
@@ -13,10 +18,11 @@ pub fn calculate_base_amount_with_fee(
     tb: &impl TransactionBuilder,
     consensus_params: &ConsensusParameters,
     previous_base_amount: u64,
-) -> u64 {
+) -> Result<u64> {
     let transaction_fee = tb
-        .fee_checked_from_tx(consensus_params)
-        .expect("Error calculating TransactionFee");
+        .fee_checked_from_tx(consensus_params)?
+        // TODO: use better error
```
Comment (digorithm) [inline packages/fuels-accounts/src/accounts_utils.rs:24]: I think this is worth creating a separate issue and linking it here.

**Comment by digorithm** [comment]
Renaming the PR just to make it visible that we're changing some public-facing methods.

File: packages/fuels/tests/providers.rs (hunk 1 above)
Comment (hal3e) [reply packages/fuels/tests/providers.rs:521]: No I just forgot to uncomment that.

File: packages/fuels-core/src/types/transaction_builders.rs [hunk 3]
```diff
@@ -188,6 +186,7 @@ macro_rules! impl_tx_trait {
 
                 Ok(num_witnesses as u8)
             }
+
```
Comment (MujkicA) [inline packages/fuels-core/src/types/transaction_builders.rs:189]:
```suggestion
```

File: packages/fuels-accounts/src/accounts_utils.rs (hunk 2 above)
Comment (hal3e) [reply packages/fuels-accounts/src/accounts_utils.rs]: I also forgot to delete this.

**Review by segfault-magnet — APPROVED** [review]

**Review by Salka1988 — APPROVED** [review]

**Review by MujkicA — APPROVED** [review]

**Review by digorithm — APPROVED** [review]
Great stuff, guys. Thanks for addressing this so quickly!

**Pull request merged**